spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/security/FirebaseAuthenticationProjectIdEvaluator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.security;
+
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+
+/**
+ *
+ * @author Felipe M Amaral
+ * @since 1.2.4
+ */
+public class FirebaseAuthenticationProjectIdEvaluator {
```

Comment:
Do we really need this class? In every other module we implement this feature directly in the auto-configuration class with code that looks something like this:
```java
	public GcpPubSubAutoConfiguration(GcpPubSubProperties gcpPubSubProperties,
			GcpProjectIdProvider gcpProjectIdProvider,
			CredentialsProvider credentialsProvider) throws IOException {
		this.gcpPubSubProperties = gcpPubSubProperties;
		this.finalProjectIdProvider = (gcpPubSubProperties.getProjectId() != null)
				? gcpPubSubProperties::getProjectId
				: gcpProjectIdProvider;
```

Reply:
Hi @meltsufin, 

I thought about this option when I started, however, I also noticed there are no unit tests for all similar places, so I applied a testable class.

I agree that this option would be simpler though.

Please let me know your thoughts

Reply:
Since we have many modules in the project, I generally encourage consistency to make things more readable and maintainable. If something can be improved, then I would do it across the board. In this particular case, I would go with the existing approach, however flawed.
Even though we might be missing some tests, I believe it should still be possible to test the existing approach.

Reply:
Fair enough. I will change to the simpler approach then. Thanks for the feedback.